src/core/lib/security/transport/security_connector.h

```diff
@@ -106,12 +98,12 @@ void grpc_security_connector_unref(grpc_security_connector *policy);
 #endif
 
 /* Check the peer. Callee takes ownership of the peer object.
-   The callback will include the resulting auth_context. */
+   Sets *auth_context and invokes on_peer_checked when done. */
 void grpc_security_connector_check_peer(grpc_exec_ctx *exec_ctx,
                                         grpc_security_connector *sc,
                                         tsi_peer peer,
-                                        grpc_security_peer_check_cb cb,
-                                        void *user_data);
+                                        grpc_auth_context **auth_context,
```

Comment:
ah, I see. but that seems a little clumsy right? IMO, the callback should have an `grpc_auth_context*` as a parameter that the implementer of the callback takes ownership of. This way, you don't have to pass it in this function.